src/Shared/CommunicationsUtilities.cs

```diff
@@ -1084,4 +1106,117 @@ public IEnumerable<KeyValuePair<string, int>> EnumerateComponents()
 
         public override string ToString() => $"{options} {salt} {fileVersionMajor} {fileVersionMinor} {fileVersionBuild} {fileVersionPrivate} {sessionId}";
     }
+
+    /// <summary>
+    /// Represents task host parameters in a structured format with named fields.
+    /// Replaces Dictionary&lt;string, string&gt; for better performance by avoiding allocation.
+    /// </summary>
+    internal readonly struct TaskHostParameters
```

Comment:
use it across codebase instead of private IDictionary<string, string> _taskHostParameters;